examples/src/test/java/com/glegoux/grpc/examples/hello/service/HelloServiceTest.java

```diff
@@ -0,0 +1,46 @@
+package com.glegoux.grpc.examples.hello.service;
+
+import com.glegoux.examples.hello.HelloReply;
+import com.glegoux.examples.hello.HelloRequest;
+import io.grpc.stub.StreamObserver;
+import org.assertj.core.api.Assertions;
+import org.junit.Test;
+
+import java.util.concurrent.atomic.AtomicInteger;
+
+public class HelloServiceTest {
+
+    AtomicInteger counter = new AtomicInteger();
+    AtomicInteger counter2 = new AtomicInteger();
+
+    StreamObserver< HelloReply > responseObserver = new StreamObserver<HelloReply>() {
+        @Override
+        public void onNext(HelloReply value) {
+            counter.incrementAndGet();
+        }
+
+        @Override
+        public void onError(Throwable t) {
+            throw new RuntimeException();
+        }
+
+        @Override
+        public void onCompleted() {
+            counter2.incrementAndGet();
+        }
+    };
+
+    @Test
+    public void test() {
+        //GIVEN
+        HelloService helloService = new HelloService();
+        HelloRequest req = HelloRequest.newBuilder().build();
+
+        //WHEN
+        helloService.sayHello(req, responseObserver);
+
+        //THEN
+        Assertions.assertThat(counter.get()).isEqualTo(1);
```

Comment:
Instead of having an atomic counter, why do you test only the value of the answer?

Reply:
modified. I implemented the test in another way but still I am not really satisfied: the bad thing is that we need to introduce variables inside the test class, just for the test.  If you have better ideas on how to test it, let me know.